superset/connectors/druid/models.py

```diff
@@ -262,6 +272,17 @@ def name(self):
     def unique_name(self):
         return self.verbose_name if self.verbose_name else self.cluster_name
 
+    def get_kerberos_auth(self):
+        enable_kerberos_authentication = conf.get('ENABLE_KERBEROS_AUTHENTICATION', False)
+        kerberos_keytab = conf.get('KERBEROS_KEYTAB', [])
+        kerberos_principal = conf.get('KERBEROS_PRINCIPAL', [])
+        if not enable_kerberos_authentication:
+            return False
+
+        kerberos_commands = 'kinit -k -t %s %s' % (kerberos_keytab, kerberos_principal)
+        subprocess.call(kerberos_commands, shell=True)
```

Comment:
Not sure I feel comfortable shelling out like this, from within a model, no less. This sort of logic belongs in a higher layer (connection manager, perhaps?)